File: contest-management-client-server-cs/networking/dto/OrganiserDTO.cs
#nullable disable
using System;
using model;

namespace networking.dto
{
    [Serializable]
    public class OrganiserDTO
    {
        public string Username { get; set; }

        public string Password { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public static explicit operator Organiser(OrganiserDTO organiserDto)
        {
            return new()
            {
                Username = organiserDto.Username,
                Password = organiserDto.Password,
                FirstName = organiserDto.FirstName,
                LastName = organiserDto.LastName
            };
        }

        public static explicit operator OrganiserDTO(Organiser organiser)
        {
            return new()
            {
                Username = organiser.Username,
                Password = organiser.Password,
                FirstName = organiser.FirstName,
                LastName = organiser.LastName
            };
        }
    }
}

File: contest-management-client-server-java/ContestMgmtNetworking/src/main/java/contestmgmt/networking/dto/OrganiserDTO.java
package contestmgmt.networking.dto;

import java.io.Serializable;

public class OrganiserDTO implements Serializable {
    private String username;

    private String password;

    public String getUsername() {
        return username;
    }

    public OrganiserDTO(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public String toString() {
        return "OrganiserDTO { username: %sn, password: %s }".formatted(username, password);
    }
}
